File: Array&String/HashTable/CPP/count_largest_group.cpp
#include <algorithm>
#include <map>
using namespace std;

class Solution {
public:
    int countLargestGroup(int n) {
        map<int, int> digitGroup;
        int maxd = 0, ans = 0;
        for (int i = 1; i < n+1; i++) {
            int sumDigits = 0;
            int cur = i;
            while (cur > 0) {
                sumDigits += cur % 10;
                cur /= 10;
            }
            digitGroup[sumDigits]++;
            maxd = max(maxd, digitGroup[sumDigits]);
        }
        for (auto it = digitGroup.begin(); it != digitGroup.end(); ++it) {
            if (it->second == maxd) ans++;
        }
        return ans;
    }
};

File: Array-String/HashTable/Python/count_largest_group.py
from collections import defaultdict

class Solution:
    def sumDigits(self, n: int) -> int:
        ans = 0
        while n:
            ans += n % 10
            n //= 10
        return ans

    def countLargestGroup(self, n: int) -> int:
        d = defaultdict(int)
        maxd = 0
        for i in range(1, n+1):
            sd = self.sumDigits(i)
            d[sd] += 1
            maxd = max(maxd, d[sd])
        res = 0
        for k in d:
            if d[k] == maxd: res += 1
        return res
